homeassistant/components/nfandroidtv/config_flow.py

```diff
@@ -32,16 +29,22 @@ def __init__(self) -> None:
 
     async def async_step_dhcp(self, discovery_info: DhcpServiceInfo) -> FlowResult:
         """Handle dhcp discovery."""
-        self.ip_address = discovery_info[IP_ADDRESS]
-        mac = format_mac(discovery_info[MAC_ADDRESS])
+        self.ip_address = discovery_info.ip
+        mac = format_mac(discovery_info.macaddress)
+        _LOGGER.warning(discovery_info)
 
-        existing_entry = await self.async_set_unique_id(self.ip_address)
-        if existing_entry:
+        if existing_entry := await self.async_set_unique_id(discovery_info.ip):
             self.hass.config_entries.async_update_entry(existing_entry, unique_id=mac)
             return self.async_abort(reason="already_configured")
         await self.async_set_unique_id(mac)
-        self._abort_if_unique_id_configured(updates={CONF_HOST: self.ip_address})
-        if "amazon-" in discovery_info[HOSTNAME]:
+        self._abort_if_unique_id_configured(updates={CONF_HOST: discovery_info.ip})
+        if "amazon-" in discovery_info.hostname:
+            session = async_get_clientsession(self.hass)
+            # A valid fire stick device should have this port open
+            try:
+                await session.get(f"http://{discovery_info.ip}:8009")
+            except ClientConnectorError:
+                return self.async_abort(reason="not_valid_device")
```

Comment:
Can this be moved to the library?

Reply:
I mean it could but I don't control the package. @engrbm87 will have to do it.
No answer to my previous PR: https://github.com/engrbm87/notifications_android_tv/pull/5
This also uses a port that has nothing to do with it. So I think it wouldn't even belong in that package. `8009` is the port labelled as ajp13 according to `nmap` 
https://developer.amazon.com/docs/fire-tv/dial-integration.html

Reply:
Every firestick afaik has this port open.